Sources/PluginLibrary/Descriptor.swift

```diff
@@ -56,35 +57,70 @@ public final class DescriptorSet {
 }
 
 public final class FileDescriptor {
+  public enum Syntax: String {
+    case proto2
+    case proto3
+
+    public init?(rawValue: String) {
+      switch rawValue {
+      case "proto2", "":
+        self = .proto2
+      case "proto3":
+        self = .proto3
+      default:
+        return nil
+      }
+    }
+  }
+
   public let proto: Google_Protobuf_FileDescriptorProto
-  public var name: String { return self.proto.name }
+  public var name: String { return proto.name }
+  public var package: String { return proto.package }
+
+  public let syntax: Syntax
+
+  public var dependencies: [String] { return proto.dependency }
+  public var publicDependencies: [String] { return proto.publicDependency.map { dependencies[Int($0)] } }
```

Comment:
In theory, but I'm trying not to make everything lazy and let some things be computed when I figured the generators can do some caching themselves.

Since imports cause the graph to get walked from one node to a completely different one, it isn't easy for a generator to wrap them with caching without rebuilding the whole graph themselves.  So I'm trying to make the graph building and the Swift types the things that end up cached so avoid generating them over an over if a type is reused in different places.